### PR TITLE
SDI-360 Update Nomis from incentive creation domain event

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: helm_deploy/hmpps-prisoner-to-nomis-update/values.yaml [hunk 1]
```diff
@@ -48,6 +48,8 @@ generic-service:
       MAPPING_CLIENT_SECRET: "MAPPING_CLIENT_SECRET"
       VISITS_CLIENT_ID: "VISITS_CLIENT_ID"
       VISITS_CLIENT_SECRET: "VISITS_CLIENT_SECRET"
+#      INCENTIVES_CLIENT_ID: "INCENTIVES_CLIENT_ID"
+#      INCENTIVES_CLIENT_SECRET: "INCENTIVES_CLIENT_SECRET"
```
Comment (steverendell) [inline helm_deploy/hmpps-prisoner-to-nomis-update/values.yaml:52]: not sure if these are needed yet

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt [hunk 2]
```diff
@@ -39,6 +39,14 @@ class NomisApiService(@Qualifier("nomisApiWebClient") private val webClient: Web
...
       .block()
   }
+
+  fun createIncentive(request: CreateIncentiveDto) =
+    webClient.post()
+      .uri("//incentives/${request.bookingId}/incentives")
+      .bodyValue(request)
+      .retrieve()
+      .bodyToMono(CreateIncentiveResponseDto::class.java)
+      .block()!!
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt:49]: This endpoint doesnt exist yet so TBD

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesApiService.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  val bookingId: Long,
+  val sequence: Long,
+  val iepDate: LocalDate,
+  val iepTime: LocalDateTime,
+  val agencyId: String,
+  val locationId: String? = null,
+  val userId: String?,
+  val reviewType: ReviewType? = null,
+  val auditModuleName: String? = null,
+)
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesApiService.kt:39]: I'm assuming these are placeholder fields, they don't look like the incentives API

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesService.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        .plus(Pair("nomisIncentiveSequence", nomisResponse.nomisIncentiveSequence.toString()))
+
+      try {
+        mappingService.createIncentiveMapping(
+          IncentiveMappingDto(
+            nomisBookingId = nomisResponse.nomisBookingId,
+            nomisIncentiveSequence = nomisResponse.nomisIncentiveSequence,
+            incentiveId = event.incentiveId,
+            mappingType = "ONLINE",
+          )
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesService.kt:77]: mapping types are MIGRATED, INCENTIVE_CREATED, NOMIS_CREATED

Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesService.kt:77]: 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesApiService.kt (hunk 3 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesApiService.kt:39]: Should be .. is this the right endpoint -> https://incentives-api-dev.hmpps.service.justice.gov.uk/webjars/swagger-ui/index.html#/iep-level-resource/getReviewById

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesApiService.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.web.reactive.function.client.WebClient
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Service
+class IncentivesApiService(@Qualifier("incentivesApiWebClient") private val webClient: WebClient) {
+
+  fun getIncentive(incentiveId: Long): IepDetail {
+    return webClient.get()
+      .uri("/iep/reviews/id/{id}/$incentiveId")
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/incentives/IncentivesApiService.kt:14]: typo here - sorted locally

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**